src/phylum/init/sig.py

```diff
@@ -1,118 +1,73 @@
-"""Helper functions for verifying minisign signatures.
+"""Verify Phylum generated digital signatures.
 
-This module is meant to be a quick and dirty means of verifying minisign signatures in Python.
-There is no readily accessible Python library at this time. The `py-minisign` repository exists
-on GitHub to attempt this - https://github.com/x13a/py-minisign - but it does not exist as a
-package on PyPI and also does not appear to be actively maintained. There is a `minisign` package
-on PyPI - https://pypi.org/project/minisign/ - but it comes from a different repo and has no
-functionality at the time of this writing.
+This module is meant to be a quick and dirty means of verifying RSA signatures in Python.
+It makes use of the hazardous materials layer of the `cryptography` library, but does so
+in a way that closely follows the example documentation:
 
-Short of forking the `py-minisign` repo to maintain it and publish a package from it on PyPI,
-the actual format for signatures and public keys is simple and so is verifying signatures.
+https://cryptography.io/en/latest/hazmat/primitives/asymmetric/rsa/#verification
 
-Minisign reference: https://jedisct1.github.io/minisign/
+There are a number of assumptions:
 
-Even still, this module is NOT meant to be used as a library or general purpose minisign
-signature verification. It is purpose written to specifically verify minisign signatures that
-were created by Phylum. As such, it makes a number of assumptions:
-
-* The Minisign Public Key for Phylum, Inc. will not change between releases
+* The RSA Public Key for Phylum, Inc. will not change between releases
+* The RSA signature was created with the SHA256 hash function
+* The RSA signature was created with the PKCS1 v1.5 padding scheme
 * The files to be verified were created by Phylum, Inc.
-* The `.minisig` signature includes a trusted comment and will therefore contain a known number of lines
-* The source of the `.minisig` signature is a trusted location, controlled by Phylum, Inc. for it's CLI releases
+* The source of the `.signature` files is a trusted location, controlled by Phylum, Inc. for it's CLI releases
```

Comment:
What if these assumptions are not upheld? I'd assume we don't fail deadly and instead it would just fail verification? If so we might want to state that to communicate to future readers that failure would still be safe.

After all if we assume the signature doesn't change and that fails deadly, it would defeat the entire point of validating signatures.

That said, the answer might not be the same to all of these assumptions so it might be necessary to clarify. For example "signature is always the same" should fail safely, while "signature is at a trusted location" would likely fail deadly.

Reply:
Good points. The language can be clarified a bit. The assumptions are more for the future maintainers of the package. That is, if and when the signature verification method changes in the future and failures start to appear, there will be bread crumbs identifying the possible causes of failure. Since the public key has been hard coded in the `sig` module, there should be no "deadly" failures. Signature verification would fail and the application will exit with a message and non-zero code. A functional test was added to check that the hard-coded signature matches the one hosted at https://raw.githubusercontent.com/phylum-dev/cli/main/scripts/signing-key.pub since that is where the [quickstart documentation](https://docs.phylum.io/docs/quickstart) directs CLI users.